src/UhdmAst.cpp

```diff
@@ -2538,9 +2538,28 @@ namespace UhdmAst {
                                   arguments[3]);
           }
         } else if (objectName == "$error") {
-          //TODO: Revisit argument handling
-          bool maybe = arguments.size() ? false : true;
-          return new AstStop(new FileLine("uhdm"), maybe);
+          AstNode* args = nullptr;
+          for (auto a : arguments) {
+            if (args == nullptr)
+              args = a;
+            else
+              args->addNextNull(a);
+          }
+          node = new AstDisplay(new FileLine("uhdm"),
+                                AstDisplayType::DT_ERROR,
+                                nullptr,
+                                args);
+          auto* stop = new AstStop(new FileLine("uhdm"), true);
+          node->addNext(stop);
+          return node;
+        } else if (objectName == "$fatal") {
+          node = new AstDisplay(new FileLine("uhdm"),
+                                AstDisplayType::DT_FATAL,
+                                nullptr,
+                                nullptr);
```

Comment:
A couple of questions:
* Why do we not handle arguments when calling `$fatal`?
* Shouldn't that code be common with `$error`?
* Adding `$warning` and `$info` seems to be trivial here, could we add that too?
* Does verilator generate the implicit `$finish` when `$fatal` is called? If not, shouldn't we generate a node for that too and add it to the statement list here explicitly?

Reply:
I missed one case in grammar where `$fatal` arguments were used, now this is corrected and all four calls (`$info`, `$warning`, `$error`, `$fatal`) use common handling.